### ISSUE TITLE
Failed to download inception model

### ISSUE THREAD

**Issue by DrewParallel** [tensorflow/hub#117]
I'm trying to run the retraining program and getting this error:

INFO:tensorflow:Using /tmp/tfhub_modules to cache modules.
INFO:tensorflow:Downloading TF-Hub Module 'https://tfhub.dev/google/imagenet/inception_v3/feature_vector/1'.
Traceback (most recent call last):
  File "retrain.py", line 1347, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "retrain.py", line 1021, in main
    module_spec = hub.load_module_spec(FLAGS.tfhub_module)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/native_module.py", line 99, in load_module_spec
    path = compressed_module_resolver.get_default().get_module_path(path)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/resolver.py", line 385, in get_module_path
    return self._get_module_path(handle)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/resolver.py", line 467, in _get_module_path
    return resolver.get_module_path(handle)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/resolver.py", line 385, in get_module_path
    return self._get_module_path(handle)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/compressed_module_resolver.py", line 105, in _get_module_path
    self._lock_file_timeout_sec())
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/resolver.py", line 313, in atomic_download
    download_fn(handle, tmp_dir)
  File "/net/home/drew/tf_venv/lib/python2.7/site-packages/tensorflow_hub/compressed_module_resolver.py", line 101, in download
    response = url_opener.open(request)
  File "/usr/lib64/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/usr/lib64/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/usr/lib64/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 1224, in https_open
    return self.do_open(httplib.HTTPSConnection, req)
  File "/usr/lib64/python2.7/urllib2.py", line 1186, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno 8] _ssl.c:504: EOF occurred in violation of protocol>

Note that running via wget works:
wget 'https://tfhub.dev/google/imagenet/inception_v3/feature_vector/1' -O /tmp/image_module.tar.gz
--2018-07-24 14:04:29--  https://tfhub.dev/google/imagenet/inception_v3/feature_vector/1
Resolving tfhub.dev (tfhub.dev)... 216.239.34.21, 216.239.36.21, 216.239.38.21, ...
Connecting to tfhub.dev (tfhub.dev)|216.239.34.21|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://www.tensorflow.org/hub/modules/google/imagenet/inception_v3/feature_vector/1 [following]
--2018-07-24 14:04:29--  https://www.tensorflow.org/hub/modules/google/imagenet/inception_v3/feature_vector/1
Resolving www.tensorflow.org (www.tensorflow.org)... 216.58.192.14, 2607:f8b0:4005:801::200e
Connecting to www.tensorflow.org (www.tensorflow.org)|216.58.192.14|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 208562 (204K) [text/html]
Saving to: ‘/tmp/image_module.tar.gz’

100%[=======================================================================================================================================>] 208,562     1.22MB/s   in 0.2s   

2018-07-24 14:04:30 (1.22 MB/s) - ‘/tmp/image_module.tar.gz’ saved [208562/208562]

**Comment by vbardiovskyg** [comment]
Hi,

is this a reproducible issue, i.e. happens every time?

**Comment by DrewParallel** [comment]
Yes, happens every time.

**Comment by andresusanopinto** [comment]
This looks like an issue with urllib2 provided by your python instalation.

Does this fails?
```
$ python2  -c "import urllib2; urllib2.urlopen('https://tfhub.dev/').read()"
```

What OS and python version do you have? Would it work if you update?

**Comment by DrewParallel** [comment]
Yes, it fails:
$ python2  -c "import urllib2; urllib2.urlopen('https://tfhub.dev/').read()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib64/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib64/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/usr/lib64/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/usr/lib64/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 1224, in https_open
    return self.do_open(httplib.HTTPSConnection, req)
  File "/usr/lib64/python2.7/urllib2.py", line 1186, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno 8] _ssl.c:504: EOF occurred in violation of protocol>

Python version is 2.7.5

OS is Linux centos

**Comment by andresusanopinto** [comment]
I wonder if this is some TLS version issue and if it will go away if you update your system (python 2.7.5 was released in 2013).

This link seems relevant:
 http://pyfound.blogspot.com/2017/01/time-to-upgrade-your-python-tls-v12.html
and has some instructions on how to debug what tls version your system supports.

Please let us know how it goes to help others in similar situation.

**Comment by DrewParallel** [comment]
I believe it is the python version. On another machine, with python 2.7.15, it works. Unfortunately, I am unable to confirm on the original machine at this time.

**Comment by sivakotich** [comment]
Is there any solution available for 2.7.6 versions? or upgrade is the only solution?

**Comment by arnoegw** [comment]
Thanks for this report. The TLS problem with old 2.7.x is now covered by the new docs page
https://github.com/tensorflow/hub/blob/master/docs/common_issues.md

The recommended solution (and the only one we know) is to upgrade.
